tasks/human_eval_076.rs

```diff
@@ -5,11 +5,44 @@ HumanEval/76
 /*
 ### VERUS BEGIN
 */
+use vstd::arithmetic::logarithm::log;
+use vstd::arithmetic::power::pow;
 use vstd::prelude::*;
-
 verus! {
 
-// TODO: Put your solution (the specification, implementation, and proof) to the task here
+fn is_simple_power(x: u32, n: u32) -> (ret: bool)
+    requires
+        n > 1,
+    ensures
+        ret <==> x == pow(n as int, log(n as int, x as int) as nat),
+{
+    let maybe_x = n.checked_pow(x.ilog(n));
+    if (maybe_x.is_some() && maybe_x.unwrap() == x) {
+        true
+    } else {
+        false
+    }
+}
+
+// note that the order of paramters is reverse in exec fn ilog and spec fn log
+#[verifier::external_fn_specification]
+pub fn ex_ilog(x: u32, base: u32) -> (ret: u32)
+    requires
+        base > 1,
```

Comment:
The Rust [docs](https://doc.rust-lang.org/std/primitive.u32.html#method.ilog) seem to imply that we need `x > 0` here too, in order to avoid a panic.

Reply:
Thanks for catching that! I've change it to work for only positive integers. The python solution doesn't account for zero and supporting zero requires deviating from a pow/log based spec

Reply:
Makes sense.